File: src/Application/Models/TravelPlan.cs
namespace Application.Models;

public class TravelPlan
{
    public string? Origin { get; set; }
    public string? Destination { get; set; }
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
}

public class TravelPlanSummary(TravelPlan plan)
{
    public bool HasOrigin { get; set; } = !string.IsNullOrEmpty(plan.Origin);
    public bool HasDestination { get; set; } = !string.IsNullOrEmpty(plan.Destination);
    public bool HasDates { get; set; } = plan.StartDate.HasValue && plan.EndDate.HasValue;
}

File: ui/src/features/travel-planning/domain/TravelPlan.ts
export interface TravelPlan {
    id: string;
    destination: string;
    startDate: Date;
    endDate: Date;
    origin: string;
    flightPlan?: FlightPlan;
}

export interface FlightPlan {
    flightOption?: FlightOption | null;
}

export interface FlightOption {
    airline: string;
    flightNumber: string;
    departure: FlightEndpoint;
    arrival: FlightEndpoint;
    duration: string;
    price: Price;
}

export interface FlightEndpoint {
    airport: string;
    airportCode: string;
    datetime: Date;
}

export interface Price {
    amount: number;
    currency: string;
}